front/pages/api/w/[wId]/files/[fileId].ts

```diff
@@ -0,0 +1,251 @@
+            api_error: {
+              type: "file_not_found",
+              message: "File not found.",
+            },
+          });
+        });
+
+        readStream.pipe(res);
+        return;
+      }
+
+      // Redirect to a signed URL.
+      const url = await fileRes.getSignedUrlForDownload(auth, "original");
+
+      res.redirect(url);
+      return;
+    }
+
+    case "DELETE": {
+      const deleteRes = await fileRes.delete(auth);
+      if (deleteRes.isErr()) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: "Failed to delete the file.",
+          },
+        });
+      }
+
+      res.status(204).end();
+      return;
+    }
+
+    case "POST": {
+      if (fileRes.isReady || fileRes.isFailed) {
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message:
+              "The file has already been uploaded or the upload has failed.",
+          },
+        });
+      }
+
+      if (
+        fileRes.createdAt.getTime() + UPLOAD_DELAY_AFTER_CREATION_MS <
+        Date.now()
+      ) {
+        await fileRes.markAsFailed();
+
+        return apiError(req, res, {
+          status_code: 400,
+          api_error: {
+            type: "invalid_request_error",
+            message: "File upload has expired. Create a new file.",
+          },
+        });
+      }
+
+      try {
+        const form = new IncomingForm({
+          // Stream the uploaded document to the cloud storage.
+          fileWriteStreamHandler: () => {
+            return fileRes.getWriteStream(auth, "original");
+          },
+
+          // Support only one file upload.
+          maxFiles: 1,
+
+          // Validate the file size.
+          maxFileSize: fileRes.fileSize,
+
+          // Ensure the file is of the correct type.
+          filter: function (part) {
+            if (part.mimetype !== fileRes.contentType) {
+              return false;
+            }
+
+            return true;
+          },
+        });
+        const [, files] = await form.parse(req);
+
+        const maybeFiles = files.file;
+
+        if (!maybeFiles || maybeFiles.length === 0) {
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "file_type_not_supported",
+              message: "File is not supported.",
+            },
+          });
+        }
+
+        const preProcessingRes = await maybeApplyPreProcessing(auth, fileRes);
+        if (preProcessingRes.isErr()) {
+          return apiError(req, res, {
+            status_code: 400,
+            api_error: {
+              type: "invalid_request_error",
+              message: "Failed to process the file.",
+            },
+          });
+        }
+
+        res.status(200).json({ file: fileRes.toJSON(auth) });
+        return;
+      } catch (error) {
```

Comment:
catch should be around external functions only. Let's narrow the catch to the relevant call. If that can come from maybeApplyPreProcessing then this catch is not valid